### DISCUSSION TITLE
Meow affecting binding to mode-specific-map in custom leader map?

### DISCUSSION THREAD

**Discussion by okamsn** [meow-edit/meow#461] [General]
I have adjusted `meow-keypad-leader-dispatch` and `meow-keymap-alist` to point to a custom leader keymap. In this custom keymap, I bind "i" to `mode-specific-map`.

So far, when I try something like `SPC i g` to mean `C-c g`, Meow intercepts this as if I had pressed `KEYPAD g` for `C-M-`. Pressing `SPC g` correctly becomes `C-M-`. Pressing `SPC i SPC g` translates to `C-c g`.

Besides changing `meow-keypad-leader-dispatch` and `meow-keymap-alist`, what else needs to be done to stop `meow-keypad` from intercepting presses for `mode-specific-map`?

**Comment by DogLooksGood** [comment]
The default behavior of g is prefix next input with `C-M-`, unless you disable it, it's always translated as `C-M-`. It's specified with variable `meow-keypad-ctrl-meta-prefix`.  BTW, having another custom leader map isn't a good idea with Meow, unless you are using a distribution like doom which provides you a leader map out-of-box.

**Reply by okamsn** [reply]
~~> The default behavior of g is prefix next input with `C-M-`, unless you disable it, it's always translated as `C-M-`.~~

~~I don't see that every `g` is _always_ translated to `C-M-`. For example, if I press `SPC a g`, that works as expected and runs the bound command. I suspect that the problem with `SPC i g` is that `SPC i` is bound to `mode-specific-map`. So I ask, are `meow-keypad-leader-dispatch` and `meow-keymap-alist` the only variables that need to be changed?~~

~~The behavior is not what I expected after changing the variables, so I am asking about it.~~

~~> BTW, having another custom leader map isn't a good idea with Meow, unless you are using a distribution like doom which provides you a leader map out-of-box.~~

~~Yeah, I'm trying to combine my existing Evil leader map with the Meow keypad with minimal changes.~~

I see what you are saying now. Thank you for the information. It looks like if the keymap is detected to have a binding using `M-`, the keypad will use `g` as `C-M-`. If it doesn't detect that, then `g` works normally.

I'll update my leader map to adapt to that.

I saw that, on Emacs 29, the `(lookup-key keymap "^[")` in `meow--keypad-has-sub-meta-keymap-p` returns non-nil even when the found keys are bound to `nil`. If that is also the behavior on other Emacs versions, I would like to make a PR for treating that case as having no keys using `meta`.

**Reply by DogLooksGood** [reply]
Well, the perfect solution would be having a real simulation with some low-level translation. But I haven't figured out how to do it yet.
